hooks/checkout

```diff
@@ -13,6 +22,24 @@ check_set() {
   fi
 }
 
+copy_checkout_from_s3() {
+  local s3_url="$1"
+
+  clean_checkout_dir
+
+  local checkout
+
+  # find most recent checkout
+  checkout=$(aws s3 ls "${s3_url}/" \
+      | (sort -r -k 4 || true) \
+      | head -n1 \
+      | awk '{print $4}'
+  )
+  aws s3 cp "${s3_url}/${checkout}" "${TMP_DIR}/${checkout}"
```

Comment:
should we copy this directly into `/mnt/ephemeral/buildkite/` instead of a temp directory?

Reply:
Went ahead and copied it into the parent directory of the checkout, also noticed that the tar needs to be extracted in the parent directory and not the checkout itself.